kpops/utils/docstring.py

```diff
@@ -1,51 +1,48 @@
 import logging
 
-from pydantic import BaseModel
-
 log = logging.getLogger("docstring_utils")
 
 
-def describe_attr(name: str, docstr: str) -> str:
+def describe_attr(name: str, docstr: str | None) -> str:
     """Read attribute description from class docstring
 
     Works only with reStructuredText docstrings.
 
     :param name: Attribute name
     :type name: str
     :param docstr: Docstring from which to read. Note that the class' docstring is stored in attr `__doc__`
-    :type docstr: str
+    :type docstr: str, None
     :returns: Description of the class attribute read from the class docstring
     :rtype: str
     """
-    docstr = docstr.partition(f":param {name}:")[2]
+    if not isinstance(docstr, str):
+        return ""
 
+    docstr = docstr.partition(f":param {name}:")[2]
     docstr = _trim_description_end(docstr)
 
     return docstr
 
 
-def describe_class(class_: type[BaseModel] | str) -> str:
+def describe_class(class_docstr: str | None) -> str:
     """Return class description from its docstring
 
     Excludes parameters and return definitions
 
     Works only with reStructuredText docstrings.
 
-    :param class_: Class whose description is to be isolated or the class' docstring
-    :type class_: type[BaseModel], str
+    :param class_docstr: Class whose description is to be isolated or the class' docstring
+    :type class_docstr: type[BaseModel], str, None
     :returns: Class description taken from the class' docstring
     :rtype: str
     """
 
-    if isinstance(class_, str):
-        class_doc = _trim_description_end(class_)
-    else:
-        class_doc = _trim_description_end(class_.__doc__)
+    class_docstr = _trim_description_end(class_docstr)
 
-    return class_doc
+    return class_docstr
```

Comment:
I would suggest handling the `None` case as an early return here

this way you don't have to deal with it in `_trim_description_end`

Reply:
True